pkg/accounting/accounting.go

```diff
@@ -195,10 +226,13 @@ func (a *Accounting) getIncreasedExpectedDebt(peer swarm.Address, accountingPeer
 	return new(big.Int).Add(expectedDebt, additionalDebt), currentBalance, nil
 }
 
-func (a *Accounting) PrepareCredit(peer swarm.Address, price uint64, originated bool) (Action, error) {
+func (a *Accounting) PrepareCredit(ctx context.Context, peer swarm.Address, price uint64, originated bool) (Action, error) {
 	accountingPeer := a.getAccountingPeer(peer)
 
-	accountingPeer.lock.Lock()
+	if err := accountingPeer.lock.TryLock(ctx); err != nil {
+		a.logger.Errorf("prepare credit: failed to acquire lock: %v", err)
+		return nil, err
```

Comment:
this would be a good place to add a metric to monitor lock contention

Reply:
and also inside the other Prepare function

Reply:
it would be difficult to measure lock contention per-se... that's what you should use the mutex/block profile for. the metric would not give much more info here, because it would just be an aggregate of those values. you can measure the amount of time spend for the entire function vs. time spent under lock in a similar manner like what we do in the localstore, but it is a very different use case - if there are a lot of goroutines waiting you'll have astronomical values for waiting (time for all of them in aggregate) vs very low values under lock (since only some will actually execute the code)